props/prop_base.h

```diff
@@ -612,14 +617,15 @@ class PropBase : CommandParser, Looper, protected SaberBase {
   }
 #else
   void PollScanId() {}
-#endif
+#endif // BLADE_ID_SCAN_MILLIS
 
   // Called from setup to identify the blade and select the right
   // Blade driver, style and sound font.
-  void FindBlade() {
-    size_t best_config = FindBestConfig();
+  void FindBlade(bool announce = false) {
+    size_t best_config = FindBestConfig(announce);
     PVLOG_STATUS << "blade = " << best_config << "\n";
     current_config = blades + best_config;
+    bool bladestatus = id() < NO_BLADE; 
```

Comment:
id() is expensive, so let's not call it again, current_config->id < NO_BLADE is better.

Reply:
I updated prop_base to do this part, it will probably cause some conflicts with this change though.
Please let me know if you need help resolving.